lib/jnpr/junos/utils/sw.py

```diff
@@ -270,44 +270,82 @@ def _issu_requirement_validation(self):
+                gres_status.group(1).lower() == 'on'):
+            self.log('Requirement FAILED: Graceful switchover status is not On')
+            return False
+        self.log('Graceful switchover status is On')
+        return True
+
+    def _issu_nssu_requirement_validation(self):
+        """
+        Checks:
+            * Check GRES is enabled
+            * Check NSR is enabled
+            * Check commit synchronize is enabled
+            * Verify that NSR is configured on the master Routing Engine
+                by using the "show task replication" command.
+
+        :returns:
+            * ``True`` if validation passes.
+            * * ``False`` otherwise
+        """
         self.log('Checking GRES status')
         conf = self._dev.rpc.get_config(filter_xml=etree.XML(
             '<configuration><chassis><redundancy><graceful-switchover/></redundancy></chassis></configuration>'),
-            options={'database': 'committed', 'inherit': 'inherit', 'commit-scripts': 'apply'})
+            options={'database': 'committed', 'inherit': 'inherit',
+                     'commit-scripts': 'apply'})
         if conf.find('chassis/redundancy/graceful-switchover') is None:
             self.log('Requirement FAILED: GRES is not Enabled in configuration')
             return False
-        self.log('Checking NSR status')
-        conf = self._dev.rpc.get_config(filter_xml=etree.XML(
-            '<configuration><routing-options><nonstop-routing/></routing-options></configuration>'),
-            options={'database': 'committed', 'inherit': 'inherit', 'commit-scripts': 'apply'})
-        if conf.find('routing-options/nonstop-routing') is None:
-            self.log('Requirement FAILED: NSR is not Enabled in configuration')
-            return False
         self.log('Checking commit synchronize status')
         conf = self._dev.rpc.get_config(
             filter_xml=etree.XML('<configuration><system><commit><synchronize/></commit></system></configuration>'),
-            options={'database': 'committed', 'inherit': 'inherit', 'commit-scripts': 'apply'})
+            options={'database': 'committed', 'inherit': 'inherit',
+                     'commit-scripts': 'apply'})
         if conf.find('system/commit/synchronize') is None:
             self.log('Requirement FAILED: commit synchronize is not Enabled in configuration')
             return False
+        self.log('Checking NSR status')
+        conf = self._dev.rpc.get_config(filter_xml=etree.XML(
+            '<configuration><routing-options><nonstop-routing/></routing-options></configuration>'),
+            options={'database': 'committed', 'inherit': 'inherit',
+                     'commit-scripts': 'apply'})
+        if conf.find('routing-options/nonstop-routing') is None:
+            self.log('Requirement FAILED: NSR is not Enabled in configuration')
+            return False
         self.log('Verifying that NSR is configured on the current Routing Engine\n'
```

Comment:
Suggest changing to: `self.log('Verifying that GRES status on the current Routing Engine\n'`